lib/app.dart

```diff
@@ -35,4 +44,44 @@ class App {
         break;
     }
   }
+
+  void initFirebase() async {
+    await fb.init('firebase/constants.json');
+    utils.setLoginDomains(fb.allowedEmailDomains);
+    fb.firebaseAuth.onAuthStateChanged.listen(_fbAuthChanged);
+    loginButton.onClick.listen((_) {
+      utils.disableLoginButton();
+      fb.signInWithGoogle();
+    });
+  }
+
+  void _fbAuthChanged(firebase.User user) async {
```

Comment:
I think this should be in the `firebase.dart` file and not in the top-level `app.dart`?